Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageMetadataView.swift

```diff
@@ -58,6 +67,9 @@ open class _ChatMessageMetadataView<ExtraData: ExtraDataTypes>: _View, UIConfigP
             timestampLabel.text = nil
         }
         currentUserVisabilityIndicator.isVisible = content?.onlyVisibleForCurrentUser ?? false
+
+        authorLabel.isHidden = content?.isSentByCurrentUser ?? true
```

Comment:
Nope, this should be decided on the `ChatMessageLayoutOptions` level. For example, in 1:1 channel, the author name is never visible, only when there are more than the current user + 1 person there, we show the names for everyone but the current user.

Reply:
Fixed, please have a look.

The fact we don't have access to `cid` from the message brings the complexity 🤔 Why not to change this:
```swift
extension _ChatChannel {
    var isDirectMessageChannel: Bool { cid.id.hasPrefix("!members") }
}
```
with:
```swift
struct _ChatMessage {
    let cid: ChannelId
}
extension ChannelId {
    var isDirectMessageChannel: Bool { id.hasPrefix("!members") }
}
```

Reply:
@evsaev gotcha. Let's leave it as it, it's not _that_ bad :) Another thing to think about properly during the audit.